src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +52,170 @@ class MissingProjectMembership extends React.Component<Props, State> {
       this.props.api,
       {
         orgId: this.props.organization.slug,
-        teamId: team.slug,
+        teamId: team.value,
       },
       {
         success: () => {
           this.setState({
             loading: false,
             error: false,
           });
+          addSuccessMessage(t('Request to join team sent.'));
         },
         error: () => {
           this.setState({
             loading: false,
             error: true,
           });
-          addErrorMessage(t('There was an error while trying to leave the team.'));
+          addErrorMessage(t('There was an error while trying to request access.'));
         },
       }
     );
   }
 
-  renderJoinTeam(team: Team, features: Set<string>) {
+  renderJoinTeam(team: string, features: Set<string>) {
+    const teamStatus = TeamStore.getBySlug(team);
```

Comment:
What I would do here is rename the usage of team a bit...

```suggestion
  renderJoinTeam(teamSlug: string, features: Set<string>) {
    const team = TeamStore.getBySlug(teamSlug);
```